packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/ModelReaderWriterContextDefinition.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.IO;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+
+namespace Microsoft.TypeSpec.Generator.ClientModel.Providers
+{
+    internal class ModelReaderWriterContextDefinition : TypeProvider
+    {
+        internal static string s_name = $"{ScmCodeModelGenerator.Instance.TypeFactory.PrimaryNamespace.RemovePeriods()}Context";
```

Comment:
Don't we need underscores to avoid conflicts?

Reply:
For the name of the context class no, there will only be one for a library and each library will have its own namespace so fully qualified there will never be a conflict